### PR TITLE
feat(metrics): Add additional statsd metrics to measure bucket performance [INGEST-954]

### PR DESCRIPTION
**Pull request by RaduW**
This PR adds two additional histogram to better measure bucketing performance

#skip-changelog

### CODE REVIEW

**Review by untitaker — APPROVED** [review]

File: relay-metrics/src/statsd.rs [hunk 1]
```diff
@@ -83,13 +84,27 @@ pub enum MetricHistograms {
...
+    /// measurement of the bucket size and complexity.
+    BucketRelativeSize,
 }
 
 impl HistogramMetric for MetricHistograms {
     fn name(&self) -> &'static str {
         match *self {
             Self::BucketsFlushed => "metrics.buckets.flushed",
             Self::BucketsFlushedPerProject => "metrics.buckets.flushed_per_project",
+            Self::MetricsPerBucket => "metrics.buckets.metrics_per_bucket",
```
Comment (untitaker) [inline relay-metrics/src/statsd.rs]: what if we call it "data points per bucket"?

Comment (RaduW) [reply relay-metrics/src/statsd.rs]: works for me, will change

File: relay-metrics/src/aggregation.rs [hunk 2]
```diff
@@ -139,6 +139,12 @@ impl DistributionValue {
         self.length
     }
 
+    /// Returns the size of the map used to store the distribution
+    /// This is only relevant for internal metrics
+    pub fn internal_size(&self) -> usize {
```
Comment (jan-auer) [inline relay-metrics/src/aggregation.rs:144]: Do these functions actually have to be public? If yes, then please ensure that the doc string adheres to conventions.

File: relay-metrics/src/lib.rs [hunk 3]
```diff
@@ -112,3 +112,4 @@ mod statsd;
 
 pub use aggregation::*;
 pub use protocol::*;
+pub use statsd::*;
```
Comment (jan-auer) [inline relay-metrics/src/lib.rs:115]: Please revert this, this module is supposed to be strictly internal like in all other crates. Either ensure that such metrics can be logged internally, or define them in a different place.

File: relay-server/src/actors/envelopes.rs [hunk 4]
```diff
@@ -2715,8 +2715,24 @@ impl Handler<SendMetrics> for EnvelopeManager {
...
+        let mut sizes: Vec<(usize, f64)> = Vec::new();
+        buckets.iter().for_each(|bucket| {
+            sizes.push((bucket.value.relative_size(), bucket.value.num_metrics()));
+        });
+
         let future = self
             .send_envelope(project_key, envelope, scoping, Instant::now())
+            .then(move |x| {
+                sizes.iter().for_each(|(rel_size, num_metrics)| {
+                    relay_statsd::metric!(
```
Comment (jan-auer) [inline relay-server/src/actors/envelopes.rs:2727]: Can't we log these in the aggregator flush to avoid exposing such internals from `relay-metrics`?

**Review by jan-auer — COMMENTED** [review]
@untitaker @RaduW -- Sorry that I'm late reviewing this, but I'd request some changes after the fact.

File: relay-metrics/src/aggregation.rs (hunk 2 above)
Comment (RaduW) [reply relay-metrics/src/aggregation.rs:144]: Ok, updated the docs and made it internal (it didn't need to be public).

**Pull request merged**